test/e2e_env/universal/strictinbounds/strict_inbound.go

```diff
@@ -0,0 +1,217 @@
+		}, "30s", "1s").Should(Succeed())
+	})
+
+	It("should allow all traffic when permissive mode", func() {
+		err := NewClusterSetup().
+			Install(Yaml(samples.MeshDefaultBuilder().
+				WithName(meshName).
+				WithBuiltinMTLSBackend("backend").
+				WithEnabledMTLSBackend("backend").
+				WithPermissiveMTLSBackends(),
+			)).
+			Install(MeshTrafficPermissionAllowAllUniversal(meshName)).
+			Setup(universal.Cluster)
+		Expect(err).ToNot(HaveOccurred())
+
+		dppInboundAddress := net.JoinHostPort(universal.Cluster.GetApp("test-server").GetIP(), "80")
+		serviceAddress := net.JoinHostPort(universal.Cluster.GetApp("test-server").GetIP(), "8080")
+
+		// then communication should works
+		Eventually(func(g Gomega) {
+			resp, err := client.CollectEchoResponse(
+				universal.Cluster, "demo-client", "test-server.mesh",
+			)
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(resp.Instance).To(Equal("test-server"))
+		}, "30s", "1s").Should(Succeed())
+
+		Eventually(func(g Gomega) {
+			resp, err := client.CollectEchoResponse(
+				universal.Cluster, "demo-client-not-in-mesh", dppInboundAddress,
+			)
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(resp.Instance).To(Equal("test-server"))
+		}, "30s", "1s").Should(Succeed())
+
+		Eventually(func(g Gomega) {
+			resp, err := client.CollectEchoResponse(
+				universal.Cluster, "demo-client", serviceAddress,
+			)
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(resp.Instance).To(Equal("test-server"))
+		}, "30s", "1s").Should(Succeed())
+
+		Eventually(func(g Gomega) {
+			resp, err := client.CollectEchoResponse(
+				universal.Cluster, "demo-client-not-in-mesh", serviceAddress,
+			)
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(resp.Instance).To(Equal("test-server"))
+		}, "30s", "1s").Should(Succeed())
+	})
+
+	It("should allow only traffic to specific ports when strict mode", func() {
+		err := NewClusterSetup().
+			Install(Yaml(samples.MeshDefaultBuilder().
+				WithName(meshName).
+				WithBuiltinMTLSBackend("backend").
+				WithEnabledMTLSBackend("backend"),
+			)).
+			Install(MeshTrafficPermissionAllowAllUniversal(meshName)).
+			Setup(universal.Cluster)
+		Expect(err).ToNot(HaveOccurred())
+
+		dppInboundAddress := net.JoinHostPort(universal.Cluster.GetApp("test-server").GetIP(), "80")
+		serviceAddress := net.JoinHostPort(universal.Cluster.GetApp("test-server").GetIP(), "8080")
+
+		// then
+		// communication should works only to DPP port
+		Eventually(func(g Gomega) {
+			resp, err := client.CollectEchoResponse(
+				universal.Cluster, "demo-client", "test-server.mesh",
+			)
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(resp.Instance).To(Equal("test-server"))
+		}, "30s", "1s").Should(Succeed())
+
+		// and
+		// the service port cannot be accessed
+		Eventually(func(g Gomega) {
+			_, err := client.CollectEchoResponse(
+				universal.Cluster, "demo-client-not-in-mesh", dppInboundAddress,
+			)
+			g.Expect(err).To(HaveOccurred())
```

Comment:
you should check errors with https://github.com/kumahq/kuma/blob/f6c7ece0ca5096fc1dc8002709a490bc27d173d9/test/framework/client/collect.go#L432

Checking `err` is incorrect because it's possible util code is failed, not the actual failure between client and server.